### PR TITLE
Add `authuser` param to GKE cluster link formatter in k8s plugin

### PR DESCRIPTION
**Pull request by szubster**
#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by szubster** [comment]
E2E failure seems to be unrelated to my changes

File: .changeset/new-plants-sort.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+'@backstage/plugin-kubernetes-react': patch
+'@backstage/plugin-kubernetes': patch
+---
+
+Add `authuser` search parameter to GKE cluster link formatter in k8s plugin
+
+Thanks to this, people with multiple simultaneously logged-in accounts to be automatically picked.
```
Comment (jamieklassen) [inline .changeset/new-plants-sort.md:8]: Checking my understanding here:
Suppose you're signed in with the Google auth provider in Backstage. This PR ensures that, when you click the dashboard link to a Kubernetes object in a GKE cluster, you will be viewing that dashboard with the same identity as the one currently logged in to Backstage? When you say "multiple simultaneously logged-in accounts", does that mean multiple accounts on the Google Cloud side where the dashboards are?

Comment (szubster) [reply .changeset/new-plants-sort.md:8]: Yes, it means Google side.
GCP console has annoying behaviour of redirecting to different places when you try to access resources you do not own.
And in my example, I have a different google account for email, and different specific one for google cloud.

Comment (szubster) [reply .changeset/new-plants-sort.md]: I have amended the changeset. Hopefully it's more clear now.

File: .changeset/new-plants-sort.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+---
+'@backstage/plugin-kubernetes-react': patch
+'@backstage/plugin-kubernetes': patch
+---
+
+Add `authuser` search parameter to GKE cluster link formatter in k8s plugin
+
+Thanks to this, people with multiple simultaneously logged-in accounts in GCP console to be automatically picked.
```
Comment (jamieklassen) [inline .changeset/new-plants-sort.md:8]:
```suggestion
Thanks to this, people with multiple simultaneously logged-in accounts in their GCP console will automatically view objects with the same email as the one signed in to the Google auth provider in Backstage.
```

**Review by jamieklassen — CHANGES_REQUESTED** [review]
I might clarify the release note a little more still since we didn't have an issue for this, but otherwise it looks good

**Review by jamieklassen — APPROVED** [review]

**Review by jamieklassen — CHANGES_REQUESTED** [review]
looks like force-pushing removed the changeset updates from before?

**Comment by szubster** [comment]
> looks like force-pushing removed the changeset updates from before?

Forgot to do pull on my side. Sorry. Fixed.
Forced push to hopefully fix E2E.

**Review by jamieklassen — APPROVED** [review]

**Comment by jamieklassen** [comment]
Hey @freben, seeing comments like [this](https://github.com/backstage/backstage/pull/21804#issuecomment-1851992144) it sounds like we can't get around these e2e failures, but I don't have the privileges to merge a PR with failing checks. Would it be appropriate to ask you to merge this one?

**Comment by szubster** [comment]
@freben Can we get it merged, so it lans in next release?
rebasing gave no improvement

**Comment by jamieklassen** [comment]
> rebasing gave no improvement

@benjdlambert gives an excellent explanation for the failing CI checks [here](https://github.com/backstage/backstage/pull/21878#issuecomment-1860077895) -- until the upstream issues frenic/csstype#189 and emotion-js/emotion#3136 are resolved and Backstage consumes their new releases, we will continue to see failures on all PRs.

**Comment by szubster** [comment]
Is adding resolution to main package.json for csstype package to fix E2E considered?
It would be nice to have this merged before release today :)

**Comment by szubster** [comment]
https://github.com/emotion-js/emotion/issues/3136 is fixed, and after rebase PR passed all tests.

**Pull request merged**